### DISCUSSION TITLE
2023顯學❓提示工程Prompt Engineering (PE) 💬

### DISCUSSION THREAD

**Discussion by xxrjun** [1101-GS4719-C/Python-Community-GS4719-C#1270] [General]
隨著ChatGPT的興起，大家應該越來越常聽到**Prompt Engineering**這個單字，或是看到越來越多以 _**教你如何有效與AI溝通**_ 為主題在賣課程的廣告或影片吧。甚至會看到類似誇張的標題 _**靠ChatGPT 不用寫程式也能年入百萬，提示工程師成矽谷新寵**_

最近系上課程作業有一題是

> 請以群眾外包的三大基本要素來討論ChatGPT的出現與應用過程，你怎麼看待它與Prompt Engineering?

因為我覺得蠻有趣的所以與大家分享一下Prompt Engineering以及我的看法。先講我妄下的結論

> 現在學習Prompt Engineering(PE)的價值相對較大，但隨著AI發展，PE長遠來看大概率會被淘汰，而這個長遠大概就是未來幾年。

## Prompt Engineering 簡介

Prompt Engineering提示工程是一門AI相關領域的概念或技術，透過設計和改進AI的prompt來提高AI的表現。Prompt Engineering的目標是建立高度有效和可控的AI系統，使其能夠準確、可靠地執行特定任務。

白話一點就是**讓 AI 相關產品給出更讓你滿意的答案**。

因為人類的語言其實不是完全精確的，或者說人們發展出即便不精確的對話也不會影響彼此間的溝通，然而這對於機器而言是難以理解的。所以才會出現PE這個概念。

因此PE的根本並不是調整AI模型，而是從使用者的角度去改變。也就是說如果哪天AI模型能夠做到 _看似_ 幾乎理解人類語言與要求，那就不需要PE了。

## Sam Altman的看法

另外，OpenAI的CEO Sam Altman在去年九月的採訪(2022/09/13)說道

> I don’t think we’ll still be doing prompt engineering in five years. And this’ll be integrated everywhere.
>
> 我不認為五年後我們還會做提示工程。這將被整合到任何地方。

> I think the fundamental interface will be natural language.
>
> 我認為根本的介面將是自然語言。 

原文網址: [AI for the Next Era OpenAI's Sam Altman on the New Frontiers of AI](https://greylock.com/greymatter/sam-altman-ai-for-the-next-era/)

![image](https://user-images.githubusercontent.com/40348319/233791589-4d022bfb-e530-4e84-a43b-d25e4d31be87.png)

## Summary

綜觀來說，我認為PE不會是一個長期的職業，也不會是我長期職涯規劃的主要領域。但現階段要使用ChatGPT等相關產品，我相信學習PE相關知識與技術仍然是值得的! 最後附上兩個關於PE我覺得很棒的專案 
- [dair-ai/Prompt-Engineering-Guide](https://github.com/dair-ai/Prompt-Engineering-Guide)
- [prompt-engineering/understand-prompt](https://github.com/prompt-engineering/understand-prompt)

**Comment by xxrjun** [comment]
剛想到一個例子，瀏覽器剛出現時也出現了許多「搜尋專家」，熟練各種搜尋相關的(高級)技巧，不過隨著產品不斷迭代更新、變得更加易用後，這些**專家**便不附存在了。我覺得這些搜尋專家蠻像現在的**AI溝通專家**。都是一個短期的過度角色

**Reply by chia-kai-chang** [reply]
@xxrjun 真的太專業了，而且居然可以把思考論點在時間軸上任意移動，而且移動範圍廣達1990年代，搜尋引擎剛誕生的時代。讓我佩服。我希望自己也可以具有縱橫時間與空間的思維論辯能力。

**Comment by Hahahahbla** [comment]
Prompt Engineering在當前AI技術發展階段中扮演著關鍵角色，通過設計和優化提示來提升AI系統的表現。隨著技術的進步，AI對自然語言的理解將變得更加準確和靈活，但PE仍然是一個重要的工具，幫助我們更好地利用和控制這些先進的技術。

**Comment by NcuJerryKao** [comment]
群眾外包的三大基本要素與ChatGPT
群眾外包（Crowdsourcing）的三大基本要素是：

多元參與：來自不同背景和技能的人參與任務。
任務分配：將大任務分解成小任務，讓更多人能參與。
回饋機制：參與者在完成任務後獲得相應的回報或評價。
ChatGPT的出現與應用過程
ChatGPT 及其應用在群眾外包的三大基本要素中扮演重要角色：

多元參與：ChatGPT 可被各行各業的人使用，無論是技術專家還是普通用戶，都能利用它進行不同的任務，如內容創作、技術支持等。
任務分配：ChatGPT 能夠高效地處理大規模的文本生成任務，將繁瑣的文本創作工作分散給多個用戶，提升生產力。
回饋機制：用戶可以對 ChatGPT 的回應進行評價，幫助改進模型的性能，提升最終結果的質量。
Prompt Engineering的價值與前景
Prompt Engineering (PE) 是一門通過設計和改進AI的提示來提高AI表現的技術：

短期價值：目前，PE 在使 AI 系統更高效和可靠方面有很大價值。通過精確設計提示，AI 能夠給出更準確和滿意的答案，滿足特定任務需求。
長期前景：隨著AI技術的不斷進步，AI模型將會越來越理解自然語言和人類的需求。這意味著在未來幾年內，PE 的重要性可能會逐漸減弱，因為AI將能夠自動適應和理解不精確的人類語言。
Sam Altman的看法
OpenAI的CEO Sam Altman 在2022年9月的採訪中提到：

未來五年內，提示工程將不再需要，因為AI系統將能夠自動理解自然語言，並被整合到各個領域。
根本的介面 將是自然語言，人們不再需要特別設計提示來與AI溝通【1】。
總結來說，現在學習Prompt Engineering具有實用價值，但隨著AI技術的快速發展，未來幾年內這項技能可能會被自然語言處理技術所取代。因此，對於現在的學生來說，掌握PE技術可以帶來短期的優勢，但也應該關注AI技術的長期發展趨勢。

**Comment by titan92601** [comment]
我認為Prompt Engineering（PE）是提高AI性能的重要領域，但隨著AI發展，可能會被取代。OpenAI CEO Sam Altman表示，未來五年或不再需要PE，因為自然語言將成為主要介面。雖然PE在現階段對使用ChatGPT等產品有幫助，但長遠來看可能會被淘汰。

**Comment by Shang621** [comment]
2023顯學❓提示工程（PE）是一家新興的技術公司，專注於開發智能提示和建議工具。其產品整合了先進的機器學習和自然語言處理技術，能夠根據用戶的需求和上下文，提供精準的提示和建議。PE的產品廣泛應用於各個領域，包括個人助理、智能搜索、自動編輯等，為用戶提供更加智能和高效的服務。該公司以創新和用戶體驗為核心價值，致力於為客戶創造價值，推動科技進步和社會發展。

**Comment by Shang621** [comment]
2023顯學❓提示工程（PE）是一家新興的技術公司，專注於開發智能提示和建議工具。其產品整合了先進的機器學習和自然語言處理技術，能夠根據用戶的需求和上下文，提供精準的提示和建議。PE的產品廣泛應用於各個領域，包括個人助理、智能搜索、自動編輯等，為用戶提供更加智能和高效的服務。該公司以創新和用戶體驗為核心價值，致力於為客戶創造價值，推動科技進步和社會發展。

**Comment by 710388** [comment]
不知道為什麼大家突然都在討論這個，但這個就很平常，平時就會用的方法，不知道為什麼突然熱度會起來
